galley/pkg/source/kube/dynamic/converter/converter_test.go

```diff
@@ -22,16 +22,17 @@ import (
 
 	"github.com/gogo/protobuf/proto"
 	"github.com/gogo/protobuf/types"
-	extensions "k8s.io/api/extensions/v1beta1"
-	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/apimachinery/pkg/util/intstr"
 
 	authn "istio.io/api/authentication/v1alpha1"
 	meshcfg "istio.io/api/mesh/v1alpha1"
 	"istio.io/istio/galley/pkg/meshconfig"
 	"istio.io/istio/galley/pkg/runtime/resource"
-	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/config"
+
+	extensions "k8s.io/api/extensions/v1beta1"
```

Comment:
the order is still out of wack. it should be core, third-party, internal.

Reply:
The order is no different than the old code. Also, this was done using goimports ... and the linter is not complaining.

Reply:
Hmm ... the linter doesn't seem to care. I've been assuming that groups were alphabetical and I've seen it both ways throughout the codebase. I don't see any docs regarding import style for Istio ... what rule are you referring to?

Reply:
The linter has issues due to golang developers breaking goimports several times. The general rule was to group SDK, then everything external, then "istio.io". That's a common style across go projects. We don't want to encourage 10 groups of imports randomly spaced by new lines.

Reply:
> The linter has issues due to golang developers breaking goimports several times. 

Until the linter is fixed, we can't really enforce this or even require this sort of cleanup as a gate for any PR.

> That's a common style across go projects. We don't want to encourage 10 groups of imports randomly spaced by new lines.

When you say it's "common style across go projects", where is this documented. I wasn't aware of it and I assume many others aren't either. It would be good to either have this documented directly in our docs or have a link to an external style guide (which we conform to) that makes this clear.

Reply:
There is a flag `-local` in goimport specifically made for this purpose. But I agree, I don't think we even have a code style in istio, so it's not worth enforcing something that is not documented.
I still believe it's our responsibility as project maintainer to leave the code cleaner than before us.

Reply:
> I still believe it's our responsibility as project maintainers to leave the code cleaner than before us.

Ironically, that's exactly what I thought I was doing ... I just though alphabetical was the "right" way :)